PYIC-3411: Point core dev and build at stubs-prod CiMit stub

Wynndow opened the pull request:
**To be merged in conjunction with https://github.com/alphagov/di-ipv-core-common-infra/pull/572**
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Point core dev and build at stubs-prod CiMit stub

### Why did it change

This updates the core-back template to have the build and dev environments use the CiMit stub in the stubs-prod account. It also updates the naming of some template params to be more consistent.

Devs still have the option to use their own deploy if desired - only the default has changed.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-3411](https://govukverify.atlassian.net/browse/PYIC-3411)


[PYIC-3411]: https://govukverify.atlassian.net/browse/PYIC-3411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Wynndow committed:
PYIC-3411: Point core dev and build at stubs-prod CiMit stub

This updates the core-back template to have the build and dev
environments use the CiMit stub in the stubs-prod account. It also
updates the naming of some template params to be more consistent.

Devs still have the option to use their own deploy if desired - only the
default has changed.

therealslimsladey commented on deploy/template.yaml:
The lambda:InvokeFunction resource statements also need `cimitEnvironment` in place of `environment` for the 'else' case

Wynndow replied:
Ooo good spot. Fill fix up.

Wynndow replied:
Made me notice I missed even more `environment` -> `cimitEnvironment` 😬

Wynndow committed:
PYIC-3411: Correct invokeFunction perms

We're using a new variable, `cimitEnvironment`, which needs to be
included in the perms or things will break.

Wynndow committed:
PYIC-3411: Fix the other places I missed

I missed a bunch of places where `environment` should have been
`cimitEnvironment`. Whoops.

therealslimsladey commented on deploy/template.yaml:
```diff
@@ -2051,21 +2057,21 @@ Resources:
                   - !Ref AWS::AccountId
                   - environment
```
-> `cimitEnvironment`

Wynndow replied:
How am I so bad at this? Fixed up.

therealslimsladey reviewed (changes requested):
one more, sorry

Wynndow committed:
PYIC-3411: 😬

How am I so bad at this.

therealslimsladey reviewed (approved):
lgtm
